File: apps/web/shared/data/personality-presets.ts
import type { CharacterPersonality } from '@/lib/character'

export type PresetKey = 'streamer' | 'mentor' | 'comedian' | 'philosopher' | 'tsundere' | 'supportive' | 'tactical'

export const PERSONALITY_PRESETS: Record<PresetKey, CharacterPersonality> = {
  streamer:    { warmth: 0.80, playfulness: 0.85, assertiveness: 0.60, empathy: 0.70, formality: 0.10, sarcasm: 0.30, emotionVolatility: 0.70, emotionResponsiveness: 0.85, emotionMemory: 0.40, stressBehavior: 'humor',    baselineMood: 'happy',   presetId: 'streamer' },
  mentor:      { warmth: 0.75, playfulness: 0.30, assertiveness: 0.70, empathy: 0.90, formality: 0.70, sarcasm: 0.05, emotionVolatility: 0.20, emotionResponsiveness: 0.60, emotionMemory: 0.80, stressBehavior: 'withdraw', baselineMood: 'neutral', presetId: 'mentor' },
  comedian:    { warmth: 0.65, playfulness: 0.95, assertiveness: 0.50, empathy: 0.50, formality: 0.05, sarcasm: 0.70, emotionVolatility: 0.90, emotionResponsiveness: 0.95, emotionMemory: 0.20, stressBehavior: 'humor',    baselineMood: 'hyped',   presetId: 'comedian' },
  philosopher: { warmth: 0.50, playfulness: 0.20, assertiveness: 0.65, empathy: 0.60, formality: 0.80, sarcasm: 0.15, emotionVolatility: 0.15, emotionResponsiveness: 0.40, emotionMemory: 0.90, stressBehavior: 'withdraw', baselineMood: 'neutral', presetId: 'philosopher' },
  tsundere:    { warmth: 0.20, playfulness: 0.60, assertiveness: 0.80, empathy: 0.35, formality: 0.20, sarcasm: 0.75, emotionVolatility: 0.85, emotionResponsiveness: 0.80, emotionMemory: 0.60, stressBehavior: 'confront', baselineMood: 'neutral', presetId: 'tsundere' },
  supportive:  { warmth: 0.90, playfulness: 0.40, assertiveness: 0.35, empathy: 0.95, formality: 0.30, sarcasm: 0.05, emotionVolatility: 0.25, emotionResponsiveness: 0.75, emotionMemory: 0.85, stressBehavior: 'deflect',  baselineMood: 'chill',   presetId: 'supportive' },
  tactical:    { warmth: 0.30, playfulness: 0.20, assertiveness: 0.90, empathy: 0.40, formality: 0.75, sarcasm: 0.20, emotionVolatility: 0.30, emotionResponsiveness: 0.55, emotionMemory: 0.70, stressBehavior: 'confront', baselineMood: 'neutral', presetId: 'tactical' },
}

export interface PresetMeta {
  label: string
  tagline: string
  dots: [number, number, number]
}

export const PERSONALITY_PRESET_META: Record<PresetKey | 'custom', PresetMeta> = {
  streamer:    { label: 'Streamer',    tagline: 'Live & reactive',  dots: [0.80, 0.85, 0.70] },
  mentor:      { label: 'Mentor',      tagline: 'Calm & guiding',   dots: [0.75, 0.20, 0.90] },
  comedian:    { label: 'Comedian',    tagline: 'Chaotic energy',   dots: [0.65, 0.90, 0.50] },
  philosopher: { label: 'Philosopher', tagline: 'Measured & deep',  dots: [0.50, 0.15, 0.60] },
  tsundere:    { label: 'Tsundere',    tagline: 'Sharp & cold',     dots: [0.20, 0.85, 0.35] },
  supportive:  { label: 'Supportive',  tagline: 'Warm & patient',   dots: [0.90, 0.25, 0.95] },
  tactical:    { label: 'Tactical',    tagline: 'Focused & direct', dots: [0.30, 0.30, 0.40] },
  custom:      { label: 'Custom',      tagline: 'Your config',      dots: [0.50, 0.50, 0.50] },
}

export const ALL_PRESET_KEYS: Array<PresetKey | 'custom'> = [
  'streamer', 'mentor', 'comedian', 'philosopher',
  'tsundere', 'supportive', 'tactical', 'custom',
]

File: src/Inktide.API.Soul/Inktide.API.Soul.Domain/ValueObjects/PersonalityPresets.cs
namespace Inktide.API.Soul.Domain.ValueObjects;

/// <summary>
/// Named personality presets. Loaded into the UI as one-click templates.
/// No DB table - these are static configuration values, not user data.
/// </summary>
public static class PersonalityPresets
{

    public static readonly IReadOnlyDictionary<string, PersonalitySettings> All =
        new Dictionary<string, PersonalitySettings>(StringComparer.OrdinalIgnoreCase)
        {
            ["streamer"] = new()
            {
                Warmth = 0.80f, Playfulness = 0.85f, Assertiveness = 0.60f,
                Empathy = 0.70f, Formality = 0.10f, Sarcasm = 0.30f,
                EmotionVolatility = 0.70f, EmotionResponsiveness = 0.85f, EmotionMemory = 0.40f,
                StressBehavior = "humor", BaselineMood = "happy", PresetId = "streamer",
            },
            ["mentor"] = new()
            {
                Warmth = 0.75f, Playfulness = 0.30f, Assertiveness = 0.70f,
                Empathy = 0.90f, Formality = 0.70f, Sarcasm = 0.05f,
                EmotionVolatility = 0.20f, EmotionResponsiveness = 0.60f, EmotionMemory = 0.80f,
                StressBehavior = "withdraw", BaselineMood = "neutral", PresetId = "mentor",
            },
            ["comedian"] = new()
            {
                Warmth = 0.65f, Playfulness = 0.95f, Assertiveness = 0.50f,
                Empathy = 0.50f, Formality = 0.05f, Sarcasm = 0.70f,
                EmotionVolatility = 0.90f, EmotionResponsiveness = 0.95f, EmotionMemory = 0.20f,
                StressBehavior = "humor", BaselineMood = "hyped", PresetId = "comedian",
            },
            ["philosopher"] = new()
            {
                Warmth = 0.50f, Playfulness = 0.20f, Assertiveness = 0.65f,
                Empathy = 0.60f, Formality = 0.80f, Sarcasm = 0.15f,
                EmotionVolatility = 0.15f, EmotionResponsiveness = 0.40f, EmotionMemory = 0.90f,
                StressBehavior = "withdraw", BaselineMood = "neutral", PresetId = "philosopher",
            },
            ["tsundere"] = new()
            {
                Warmth = 0.20f, Playfulness = 0.60f, Assertiveness = 0.80f,
                Empathy = 0.35f, Formality = 0.20f, Sarcasm = 0.75f,
                EmotionVolatility = 0.85f, EmotionResponsiveness = 0.80f, EmotionMemory = 0.60f,
                StressBehavior = "confront", BaselineMood = "neutral", PresetId = "tsundere",
            },
        };

}
